Clusters without bastions do not reconcile

/kind bug

**What steps did you take and what happened:**
Deployed a cluster at 0.5.3, then rolled out 0.6.1. The `capo-controller-manager` entered a crash-loop with the following panic:

```
I0429 12:36:14.055908       1 openstackcluster_controller.go:300] controller/openstackcluster "msg"="Reconciling Bastion" "cluster"="matt-dev" "name"="matt-dev" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="OpenStackCluster" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1097244]

goroutine 482 [running]:
sigs.k8s.io/cluster-api-provider-openstack/controllers.bastionToInstanceSpec(0x4000580400, {0x40005bb870, 0x8})
        /workspace/controllers/openstackcluster_controller.go:365 +0xd4
sigs.k8s.io/cluster-api-provider-openstack/controllers.deleteBastion(0x40005d6080, 0x4000956000, 0x4000580400)
        /workspace/controllers/openstackcluster_controller.go:226 +0x1bc
sigs.k8s.io/cluster-api-provider-openstack/controllers.reconcileBastion(0x40005d6080, 0x4000956000, 0x4000580400)
        /workspace/controllers/openstackcluster_controller.go:303 +0x720
sigs.k8s.io/cluster-api-provider-openstack/controllers.reconcileNormal({0x1629178, 0x4000781950}, 0x40005d6080, 0x4000145880, 0x4000956000, 0x4000580400)
        /workspace/controllers/openstackcluster_controller.go:263 +0x138
sigs.k8s.io/cluster-api-provider-openstack/controllers.(*OpenStackClusterReconciler).Reconcile(0x40007e20c0, {0x1629178, 0x4000781950}, {{{0x400030e3d0, 0x7}, {0x400030e3c8, 0x8}}})
        /workspace/controllers/openstackcluster_controller.go:128 +0x584
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x4000474000, {0x1629178, 0x40007818c0}, {{{0x400030e3d0, 0x7}, {0x400030e3c8, 0x8}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114 +0x22c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0x4000474000, {0x16290d0, 0x40004b0440}, {0x1263ac0, 0x4000596560})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311 +0x2ac
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0x4000474000, {0x16290d0, 0x40004b0440})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266 +0x1d4
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0x40005ba620, 0x4000474000, {0x16290d0, 0x40004b0440})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227 +0x60
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:223 +0x324
```

**What did you expect to happen:**
The controller to upgrade the resources.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): 0.6.1
- Cluster-API version: 1.1.3
- OpenStack version: Train
- Minikube/KIND version: N/A
- Kubernetes version (use `kubectl version`): 1.23
- OS (e.g. from `/etc/os-release`): Darwin

--- a/controllers/openstackcluster_controller.go
+++ b/controllers/openstackcluster_controller.go
@@ -221,12 +221,12 @@ func deleteBastion(scope *scope.Scope, cluster *clusterv1.Cluster, openStackClus
 				}
 			}
 		}
-	}
 
-	instanceSpec := bastionToInstanceSpec(openStackCluster, cluster.Name)
-	if err = computeService.DeleteInstance(openStackCluster, instanceSpec, instanceStatus); err != nil {
-		handleUpdateOSCError(openStackCluster, errors.Errorf("failed to delete bastion: %v", err))
-		return errors.Errorf("failed to delete bastion: %v", err)
+		instanceSpec := bastionToInstanceSpec(openStackCluster, cluster.Name)
+		if err = computeService.DeleteInstance(openStackCluster, instanceSpec, instanceStatus); err != nil {
+			handleUpdateOSCError(openStackCluster, errors.Errorf("failed to delete bastion: %v", err))
+			return errors.Errorf("failed to delete bastion: %v", err)
+		}
 	}
 
 	openStackCluster.Status.Bastion = nil

